addons/im_livechat/static/src/legacy/widgets/livechat_button.js

```diff
@@ -106,7 +107,7 @@ const LivechatButton = Widget.extend({
         });
         const message = new WebsiteLivechatMessage(this, data, options);
 
-        const hasAlreadyMessage = _.some(this._messages, function (msg) {
+        const hasAlreadyMessage = _.some(this.messaging.livechatButtonView.messages, function (msg) {
```

Comment:
```suggestion
        const hasAlreadyMessage = this.messaging.livechatButtonView.messages.some(function (msg) {
```

Reply:
I'm not 100% confident that this won't break code.
`_.some()` silently handle falsy values, whereas `.some()` **must** be a collection.

I don't really understand the code right now, so the structural changes are just some steps in the right direction, and it gives me time to understand the code, and clean the mess in modelling.

Reply:
So I made other changes, but this one I keep `_.some()`